### ISSUE TITLE
would it be possible to get tts.py working again, would really appreciate it

### ISSUE THREAD

**Issue by lonewulfa** [x0rnn/minqlx-plugins#10]

**Comment by x0rnn** [comment]
What's the error?

**Comment by lonewulfa** [comment]
Let me apologize, just now seeing this comment... There is no error to report ... the !tts command just doestn work any longer... Im thinking its just an out-dated API or URL in the code! Players really love this plugin and it's been missed is why I reached out to you here. Thanks I patiently await your response!

Plugin loads without error or exception in minqlx.log file ... no error to report

**Comment by x0rnn** [comment]
Ah, I see, the website/API is dead. I'll try to make a local version that reads from a file. I'll close this issue once I'm done so you'll know.

**Comment by lonewulfa** [comment]
Thank you so much x0rnn!!! I see the new tts.txt file... will this file just sit in the /minqlx-plugins/ directory? and we run tts.py and the workshop item like before?

Thanks again!!!

**Comment by x0rnn** [comment]
Correct, I'll try to do it today. Might not support all the words as previously though, I'm using a smaller dictionary.

**Comment by x0rnn** [comment]
Okay, I think it should work now. I don't have QL installed any longer so I can't test properly. Let me know before I close the issue.

**Comment by lonewulfa** [comment]
Ok, this is what I have for you. With the plugin myirc.py loaded (keep in mind this plugin normally loads without error, or exception) seeing error while loading tts.py and while myirc.py is loaded https://gist.github.com/lonewulfa/e9278912683ab63a624ef37ac06164ca

And then we have the ql server spinning up without trying to load the myirc.py plugin and the tts.py shows errors, exception
https://gist.github.com/lonewulfa/bb15c1dc6d37febea2f3f74fd943e5be

Do not try to load the tts.py and everything is as normal, no errors or exceptions
https://gist.github.com/lonewulfa/bdb4832740ce96e932f3a866d15a1a1b

I have the latest tts.py (did a wget on the raw file), the latest tts.txt (did a wget on the raw file, its just sitting in the /minqlx-plugins/ directory along with the tts.py and I've added 710007616 to qlx_workshopReferences and workshop.txt

Also you may want to edit the tts.py to wget the raw version of the file instead of
https://github.com/x0rnn/minqlx-plugins/blob/master/tts.py#L2

hope this helps! Patiently awaiting your response.

**Comment by x0rnn** [comment]
Oh right, it needs the absolute path. Try again, I updated it.

**Comment by lonewulfa** [comment]
tts.py plugin is loading now without error, but doing a !tts test 1 2
does not return any sounds in-game

running latest version updated like 15 minutes ago
also I just check the steam workshop tts item of yours for updates, but it has not been updated.

**Comment by x0rnn** [comment]
I updated it 12 minutes ago, be sure to update if you downloaded earlier. Try just !tts test

**Comment by lonewulfa** [comment]
(09:44:11) [DEBUG @ minqlx.dispatch] client_command(Player(9:'JUGGERTHOT':76561197960665909), 'score')
(09:44:12) [DEBUG @ minqlx.dispatch] client_command(Player(3:'pasha.borsuk':76561198257899474), 'score')
(09:44:13) [DEBUG @ minqlx.dispatch] player_spawn(Player(3:'pasha.borsuk':76561198257899474),)
(09:44:13) [DEBUG @ minqlx.dispatch] client_command(Player(9:'JUGGERTHOT':76561197960665909), 'score')
(09:44:13) [DEBUG @ minqlx.dispatch] client_command(Player(0:'lwf':76561198154317273), 'say !tts test')
(09:44:13) [DEBUG @ minqlx.tts.execute] 76561198154317273 executed: tts @ tts -> chat
(09:44:17) [DEBUG @ minqlx.handle_console_print] broadcast: print "^1RED TEAM^3 WINS the round!^7 (3 players remaining)\n"
(09:44:17) [DEBUG @ minqlx.dispatch] round_end({'MATCH_GUID': '0c2633b2-599c-40a8-aa1c-76625fd4733d', 'ROUND': 15, 'TEAM_WON': 'RED', 'T)
(09:44:18) [DEBUG @ minqlx.dispatch] client_command(Player(8:'N.W.A EazyE':76561198067260665), 'score')
(09:44:18) [DEBUG @ minqlx.dispatch] client_command(Player(9:'JUGGERTHOT':76561197960665909), 'score')
(09:44:18) [DEBUG @ minqlx.dispatch] client_command(Player(4:'Stinger':76561198080677481), 'score')

on latest version of tts.py
!tts test
does not return any sounds in-game

**Comment by lonewulfa** [comment]
let me try again... I see an update to tts.py like just a couple minutes ago... brb

**Comment by x0rnn** [comment]
Just noticed a bug, fixed. Try again xD

**Comment by lonewulfa** [comment]
(09:49:46) [DEBUG @ minqlx.dispatch] client_command(Player(0:'lwf':76561198154317273), '!reload tts')
(09:49:46) [DEBUG @ minqlx.plugin_manager.execute] 76561198154317273 executed: reload @ plugin_manager -> client_command
(09:49:46) [INFO @ minqlx.unload_plugin] Unloading plugin 'tts'...
(09:49:46) [DEBUG @ minqlx.dispatch] unload('tts',)
(09:49:46) [INFO @ minqlx.load_plugin] Loading plugin 'tts'...
(09:49:47) [DEBUG @ minqlx.handle_console_print] broadcast: print "^4BLUE TEAM^3 WINS the round!^7 (3 players remaining)\n"

on the latest version of tts.py
plugin is unloading / loading just fine I see no errors and/or exceptions here
but
!tts test
is still not returning anything (sound(s) included) from in-game

**Comment by lonewulfa** [comment]
Im not sure what all the steam workshop item for this tts.py is doing, but does that maybe need to be updated? :)

**Comment by x0rnn** [comment]
No, the workshop item doesn't have to be updated. I have no clue why it doesn't work now. I tested the code on command prompt without the game and it returns what it's supposed to, no idea why it doesn't work ingame. :/

**Comment by x0rnn** [comment]
Ok, one more fix. Try again?

**Comment by lonewulfa** [comment]
x0rnn its working now... on your latest version of tts.py

!tts test 
worked.

!tts Testing 1 2
returned the Testing sound(s)
but 1 and 2 did not return any sound(s)

!tts Testing one two
worked

!tts Hello every one
worked 

:D

**Comment by lonewulfa** [comment]
over 25,000 lines of words/letters in the dictionary you wrote... Holy Cow!

**Comment by x0rnn** [comment]
Glad it works again! Yea, numbers don't work any more, and I didn't actually write the dictionary, I just parsed the most common English words out of an existing dictionary so it isn't too big. You can add some abbreviations such as "wtf, omg, ffs, lol, rofl" yourself to tts.txt by looking up full words such as "what, the, fuck", etc. and adding them to a new line if you want. Closing this now.

**Comment by lonewulfa** [comment]
Thank you so much x0rnn <3
might play around with adding abbreviations as well

Thanks!
